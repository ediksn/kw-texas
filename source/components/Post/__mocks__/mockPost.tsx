const mockPost = {
  id: 1,
  author: {
    name: 'Nina Royals',
    picture: '__mocks__/avatarNinaRoyals.png'
  },
  content:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean tellus leo, fringilla at elit sed, tincidunt viverra dui. Nullam dignissim mi et tincidunt luctus. Nullam ut eleifend diam. Phasellus faucibus aliquam leo, vitae scelerisque nunc dignissim in. Maecenas vulputate quam id metus efficitur tristique. Ut maximus varius enim eleifend feugiat. Ut at ultricies lorem, eget molestie risus. Nam ultricies tempus sem quis fermentum. Sed a eros ac metus scelerisque hendrerit. Ut nec urna aliquam, sagittis eros non, bibendum risus. Ut congue eleifend velit eu dictum. Praesent eu mi vitae leo pulvinar ullamcorper. Sed tortor felis, tristique vitae lacinia sit amet, pellentesque ut felis. Nulla finibus erat libero, eu sodales metus auctor at. Duis iaculis leo vitae odio mollis, feugiat aliquam erat eleifend. Mauris quis dui at lorem tincidunt viverra nec vel mauris. Pellentesque laoreet dapibus purus, eget vehicula nunc scelerisque quis. Donec ac nibh sed erat consequat sagittis ut ut odio. Curabitur tempor vitae neque vel gravida. Morbi varius ligula nec tellus accumsan, quis pharetra ante dictum. Nulla pharetra ac ex id cursus. Vivamus ut lacinia nunc. Maecenas nec orci risus.',
  date: '4 hours ago',
  likes: 13,
  comments: 5,
  shares: 2
}

export default mockPost
