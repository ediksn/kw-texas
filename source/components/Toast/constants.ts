import { verticalScale } from 'react-native-size-matters'

export const TOAST_HEIGHT = verticalScale(48)
export const TOAST_TYPE = {
  ERROR: 'ERROR',
  SUCCESS: 'SUCCESS'
}
