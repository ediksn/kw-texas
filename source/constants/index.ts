export { FORM } from './form'
export { NAVIGATION } from './navigation'
export { theme } from './theme'
export { ACTION } from './action'
export * from './statis'
export * from './testIds'
