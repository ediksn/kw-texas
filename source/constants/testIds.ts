export const TEST_IDS = {
  GENERAL_COMPONENTS: {
    VIDEO_CARD_ID: 'videoCardId',
    VIDEO_LIST_ID: 'videoListId',
    EVENT_CARD_ID: 'eventCardId',
    TOAST_ID: 'toastId'
  },
  EVENT_CARD: {
    IMAGE_ID: 'eventCardImageId',
    DATE_ID: 'eventCardImageId',
    TITLE_ID: 'eventCardTitleId',
    ASSISTANCE_ID: 'eventCardAssistanceId',
    INTERESTED_BUTTON_ID: 'eventCardInterestedButtonId',
    SHARE_BUTTON_ID: 'eventCardShareButtonId'
  },
  TOAST: {
    ICON_ID: 'toastIconId',
    TITLE_ID: 'toastTitleId'
  }
}
export const usernameInput = 'username-input'
export const passwordInput = 'password-input'
export const signinButton = 'signin-button'
export const forgotButton = 'forgot-button'
export const pickPrompts = 'pickPrompts-component'
export const kwLogo = 'kw-logo'
export const illustrationLogo = 'illustration-logo'
export const avatarPost = 'avatar-post'
export const authorPost = 'author-post'
export const datePost = 'date-post'
export const dotsOptionsPost = 'dotsoptions-post'
export const contentPost = 'content-post'
export const buttonPost = 'button-post'
