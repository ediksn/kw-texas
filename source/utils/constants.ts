export const MAX_CHARACTERS_NEW_POST: number = 500
export const POST_PREVIEW_MEDIA_LENGTH: number = 3
export const GROUP_PREVIEW_MEDIA_LENGTH: number = 10
