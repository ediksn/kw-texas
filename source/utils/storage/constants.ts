export default {
  LOGIN: {
    USER: 'LOGIN.USER',
    SESSION: 'LOGIN.SESSION'
  },
  BIOMETRIC: {
    PERMISSION: 'BIOMETRIC.PERMISSION',
    CREDENTIALS: 'BIOMETRIC.CREDENTIALS'
  }
}
