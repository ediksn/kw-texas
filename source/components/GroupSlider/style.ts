import { StyleSheet } from 'react-native'
import { scale, verticalScale } from 'react-native-size-matters'
import { theme } from '~/constants'

export const styles = StyleSheet.create({
  groups: {
    flex: 1,
    flexGrow: 0.21,
    marginVertical: scale(16),
    marginLeft: scale(16)
  },
  containerHeaderGroup: {
    marginVertical: scale(9),
    marginHorizontal: scale(16),
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around'
  },
  containerText: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  headerTextGroup: {
    fontFamily: 'Mulish-Regular',
    fontSize: theme.fonts.MEDIUM_SIZE,
    fontStyle: 'normal',
    fontWeight: '700',
    letterSpacing: 0,
    textAlign: 'left',
    color: theme.lightColor
  },
  horizontalLine: {
    height: verticalScale(1),
    backgroundColor: theme.post.horizontalLine,
    opacity: 0.1
  },
  imageContainer: {
    flex: 1,
    borderRadius: 4
  },
  headerIcon: {
    marginRight: scale(10)
  },
  headerRightIcon: {
    marginRight: scale(5)
  }
})
