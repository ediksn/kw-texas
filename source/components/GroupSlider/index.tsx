import React from 'react'
import { FlatList, View, Text, TouchableOpacity, ImageBackground, TouchableWithoutFeedback } from 'react-native'
import { useTranslation } from 'react-i18next'

import { scale } from 'react-native-size-matters'
import { useNavigation } from '@react-navigation/native'
import { styles } from './style'
import Group from '~/components/Group'
import { styles as stylesGroup } from '~/components/Group/styles'

import { GroupInterface } from '~/interfaces/groupInterface'
import { GROUP_PREVIEW_MEDIA_LENGTH } from '~/utils/constants'
import { TEST_IDS } from '~/constants/testIds'
import { Icon } from '..'
import { NAVIGATION, theme } from '~/constants'

interface GroupSliderProps {
  groups: GroupInterface | any
  onPressGroup: (group: GroupInterface) => void
  onPressMoreGroups: () => void
}

const GroupSlider = ({ groups, onPressGroup, onPressMoreGroups }: GroupSliderProps) => {
  const { t } = useTranslation()
  const { navigate } = useNavigation()
  const keyExtractorGroup = (group: GroupInterface) => group.id
  const moreGroups = groups.length > GROUP_PREVIEW_MEDIA_LENGTH
  const moreGroupsCount = groups.length - GROUP_PREVIEW_MEDIA_LENGTH
  const limitGroups = [...groups.slice(0, GROUP_PREVIEW_MEDIA_LENGTH)]

  const goToInterest = () => navigate(NAVIGATION.SCREEN.GROUPSINTERESTING)

  const headerGroup = () => (
    <View style={styles.containerHeaderGroup}>
      <View style={styles.containerText}>
        <Icon name='people-icon' size={18} viewStyle={styles.headerIcon} />
        <Text style={styles.headerTextGroup}>{t('components_Home_Header_Groups')}</Text>
      </View>
      <View style={styles.containerText}>
        <TouchableWithoutFeedback onPress={goToInterest}>
          <View style={styles.containerText}>
            <Icon name='people-icon' size={18} viewStyle={styles.headerRightIcon} color={theme.darkGreenColor} />
            <Text style={[styles.headerTextGroup, { color: theme.darkGreenColor, marginRight: scale(5) }]}>
              {t('components_Group_interesting')}
            </Text>
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback>
          <View style={styles.containerText}>
            <Icon name='people-icon' size={18} viewStyle={styles.headerRightIcon} color={theme.darkGreenColor} />
            <Text style={[styles.headerTextGroup, { color: theme.darkGreenColor }]}>
              {t('components_Group_invitations')}
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </View>
  )

  const HorizontalLine = () => <View style={styles.horizontalLine} />

  const showMoreGroups = () => {
    return (
      <View style={stylesGroup.containerCount}>
        <TouchableOpacity style={stylesGroup.container} onPress={() => onPressMoreGroups()}>
          <ImageBackground
            source={{ uri: groups[GROUP_PREVIEW_MEDIA_LENGTH + 1]?.icon?.url }}
            style={stylesGroup.containerImage}
            resizeMode='cover'
            imageStyle={stylesGroup.image}
          >
            <View style={stylesGroup.backgroundCount}>
              <Text style={stylesGroup.textCount}>{`+ ${moreGroupsCount}`}</Text>
            </View>
          </ImageBackground>
        </TouchableOpacity>
      </View>
    )
  }
  const renderGroup = ({ item }: { item: GroupInterface | any }) => <Group group={item} onGroupPress={onPressGroup} />
  return (
    <>
      {headerGroup()}
      <HorizontalLine />
      <FlatList
        testID={TEST_IDS.GROUPS.GROUP_SLIDER_ID}
        renderItem={renderGroup}
        data={limitGroups}
        keyExtractor={keyExtractorGroup}
        style={styles.groups}
        showsHorizontalScrollIndicator={false}
        horizontal
        ListFooterComponent={moreGroups ? showMoreGroups() : null}
        ListFooterComponentStyle={styles.imageContainer}
      />
      <HorizontalLine />
    </>
  )
}

export default GroupSlider
