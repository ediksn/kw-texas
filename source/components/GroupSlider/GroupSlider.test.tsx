import React from 'react'
import { render, RenderAPI } from '@testing-library/react-native'
import GroupSlider from '~/components/GroupSlider/index'
import { groupsResponse } from '../../../__mocks__/mockResponses'
import { TEST_IDS } from '~/constants/testIds'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

describe('Render Group Slider Test', () => {
  let rendered: RenderAPI
  const mockOnPressGroup = jest.fn()
  const mockOnPressMoreGroups = jest.fn()

  beforeEach(() => {
    rendered = render(
      <GroupSlider groups={groupsResponse} onPressGroup={mockOnPressGroup} onPressMoreGroups={mockOnPressMoreGroups} />
    )
  })

  it('should render component Group Slider correctly', () => {
    const component = rendered.getAllByTestId(TEST_IDS.GROUPS.GROUP_SLIDER_ID)
    expect(component).not.toBeNull()
  })

  it('Match with the snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(rendered).toBeDefined()
  })
})
