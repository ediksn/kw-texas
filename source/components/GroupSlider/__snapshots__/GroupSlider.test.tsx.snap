// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Group Slider Test Match with the snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginHorizontal": 34.285714285714285,
        "marginVertical": 19.285714285714285,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 18,
            },
            Array [
              Object {
                "marginRight": 21.428571428571427,
              },
              Object {
                "color": "#000000",
              },
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#3D424D",
            "fontFamily": "Mulish-Regular",
            "fontSize": 22,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "textAlign": "left",
          }
        }
      >
        components_Home_Header_Groups
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10.714285714285714,
                },
                Object {
                  "color": "#006C84",
                },
              ],
              Object {
                "fontFamily": "icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#3D424D",
                "fontFamily": "Mulish-Regular",
                "fontSize": 22,
                "fontStyle": "normal",
                "fontWeight": "700",
                "letterSpacing": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#006C84",
                "marginRight": 10.714285714285714,
              },
            ]
          }
        >
          components_Group_interesting
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10.714285714285714,
                },
                Object {
                  "color": "#006C84",
                },
              ],
              Object {
                "fontFamily": "icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#3D424D",
                "fontFamily": "Mulish-Regular",
                "fontSize": 22,
                "fontStyle": "normal",
                "fontWeight": "700",
                "letterSpacing": 0,
                "textAlign": "left",
              },
              Object {
                "color": "#006C84",
              },
            ]
          }
        >
          components_Group_invitations
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#2b2b33",
        "height": 1.9617647058823529,
        "opacity": 0.1,
      }
    }
  />,
  <RCTScrollView
    ListFooterComponent={null}
    ListFooterComponentStyle={
      Object {
        "borderRadius": 4,
        "flex": 1,
      }
    }
    data={
      Array [
        Object {
          "description": "description",
          "icon": Object {
            "filename": "hello.jpg",
            "id": "1",
            "url": "https://avatar.kwconnect.com/556397.jpeg",
          },
          "id": "1",
          "membersCount": 1,
          "name": "Test",
          "postCount": 1,
          "status": 1,
        },
        Object {
          "description": "description",
          "icon": Object {
            "filename": "hello.jpg",
            "id": "2",
            "url": "https://avatar.kwconnect.com/556397.jpeg",
          },
          "id": "2",
          "membersCount": 2,
          "name": "Test",
          "postCount": 2,
          "status": 2,
        },
        Object {
          "description": "description",
          "icon": Object {
            "filename": "hello.jpg",
            "id": "3",
            "url": "https://avatar.kwconnect.com/556397.jpeg",
          },
          "id": "3",
          "membersCount": 3,
          "name": "Test",
          "postCount": 3,
          "status": 3,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "flexGrow": 0.21,
        "marginLeft": 34.285714285714285,
        "marginVertical": 34.285714285714285,
      }
    }
    testID="groupSliderId"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "borderRadius": 4,
              "flex": 1,
            }
          }
          testID="groupId"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 4,
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "#282B33",
                  "borderRadius": 4,
                  "flex": 1,
                  "flexDirection": "column",
                  "marginRight": 8,
                  "width": 96,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://avatar.kwconnect.com/556397.jpeg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": 96,
                    },
                    Object {
                      "borderRadius": 4,
                      "height": 96,
                      "width": 96,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(43, 43, 51, 0.6)",
                    "borderRadius": 4,
                    "flex": 1,
                  }
                }
              >
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={4}
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Mulish-Bold",
                      "fontStyle": "normal",
                      "letterSpacing": 0,
                      "lineHeight": 25.142857142857142,
                      "marginHorizontal": 17.142857142857142,
                      "marginVertical": 17.142857142857142,
                      "textAlign": "left",
                    }
                  }
                >
                  Test
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "borderRadius": 4,
              "flex": 1,
            }
          }
          testID="groupId"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 4,
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "#282B33",
                  "borderRadius": 4,
                  "flex": 1,
                  "flexDirection": "column",
                  "marginRight": 8,
                  "width": 96,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://avatar.kwconnect.com/556397.jpeg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": 96,
                    },
                    Object {
                      "borderRadius": 4,
                      "height": 96,
                      "width": 96,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(43, 43, 51, 0.6)",
                    "borderRadius": 4,
                    "flex": 1,
                  }
                }
              >
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={4}
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Mulish-Bold",
                      "fontStyle": "normal",
                      "letterSpacing": 0,
                      "lineHeight": 25.142857142857142,
                      "marginHorizontal": 17.142857142857142,
                      "marginVertical": 17.142857142857142,
                      "textAlign": "left",
                    }
                  }
                >
                  Test
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "borderRadius": 4,
              "flex": 1,
            }
          }
          testID="groupId"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 4,
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "#282B33",
                  "borderRadius": 4,
                  "flex": 1,
                  "flexDirection": "column",
                  "marginRight": 8,
                  "width": 96,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://avatar.kwconnect.com/556397.jpeg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": 96,
                    },
                    Object {
                      "borderRadius": 4,
                      "height": 96,
                      "width": 96,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(43, 43, 51, 0.6)",
                    "borderRadius": 4,
                    "flex": 1,
                  }
                }
              >
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={4}
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Mulish-Bold",
                      "fontStyle": "normal",
                      "letterSpacing": 0,
                      "lineHeight": 25.142857142857142,
                      "marginHorizontal": 17.142857142857142,
                      "marginVertical": 17.142857142857142,
                      "textAlign": "left",
                    }
                  }
                >
                  Test
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      Object {
        "backgroundColor": "#2b2b33",
        "height": 1.9617647058823529,
        "opacity": 0.1,
      }
    }
  />,
]
`;
